ui/src/app/components/rollout-widget/rollout-widget.tsx

```diff
@@ -0,0 +1,134 @@
+import * as React from 'react';
+import {Link} from 'react-router-dom';
+
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import {faCircleNotch, faRedoAlt} from '@fortawesome/free-solid-svg-icons';
+import {IconDefinition} from '@fortawesome/fontawesome-svg-core';
+import {faStar as faStarSolid} from '@fortawesome/free-solid-svg-icons';
+import {faStar as faStarOutline} from '@fortawesome/free-regular-svg-icons/faStar';
+
+import {Tooltip} from 'antd';
+
+import {ParsePodStatus, PodStatus, ReplicaSets} from '../pods/pods';
+import {RolloutInfo} from '../../../models/rollout/rollout';
+import {useWatchRollout} from '../../shared/services/rollout';
+import {useClickOutside} from '../../shared/utils/utils';
+import {InfoItemKind, InfoItemRow} from '../info-item/info-item';
+import {RolloutAction, RolloutActionButton} from '../rollout-actions/rollout-actions';
+import {RolloutStatus, StatusIcon} from '../status-icon/status-icon';
+import './rollout-widget.scss';
+
+export const isInProgress = (rollout: RolloutInfo): boolean => {
+    for (const rs of rollout.replicaSets || []) {
+        for (const p of rs.pods || []) {
+            const status = ParsePodStatus(p.status);
+            if (status === PodStatus.Pending) {
+                return true;
+            }
+        }
+    }
+    return false;
+};
+
+export const RolloutWidget = (props: {
```

Comment:
Since there is already a [component](https://github.com/argoproj/argo-rollouts/blob/6716d50547049e8017361736aef003a9ce10df86/ui/src/app/components/rollout/rollout.tsx#L92) with same name. Would it be better to rename it differently.

Reply:
@ashutosh16 I've renamed my new widget to `RolloutGridWidget`

Reply:
I'll review the PR today.